howso/analysis.amlg

```diff
@@ -15,57 +15,44 @@
-	;			the same session id index or missing session indices, therefore this flag should only be used internally for validation
-	; bypass_hyperparameter_analysis : optional
-	; bypass_calculate_feature_residuals : optional
-	; bypass_calculate_feature_weights : optional
-	; use_deviations: optional, default is null, will auto-determine whether to use deviations for LK metric in queries. When true forces the use
-	;			of deviations , when false will not use deviations.
-	; num_samples: used in calculating feature residuals
-	; k_values: optional list used in hyperparameter search
-	; p_values: optional list used in hyperparameter search
-	; dt_values: optional list used in hyperparameter search
-	; targeted_model: enumeration, default is "single_targeted"
-	;   "single_targeted" = analyze hyperparameters for the specified action_features
-	;   "omni_targeted" = analyze hyperparameters for each action feature, using all other features as context_features. if action_features aren't specified, uses all context_features.
-	;   "targetless" = analyze hyperparameters for all context features as possible action features, ignores action_features parameter
-	; num_analysis_samples : optional. number of cases to sample during analysis. only applies for k_folds = 1
-	; analysis_sub_model_size: optional. number of samples to use for analysis. the rest will be randomly held-out and not included in calculations
-	; inverse_residuals_as_weights: optional, default is null, will be set to false for targeted and true for targetless
-	;			when true will forcibly compute and use inverse of residuals as feature weights
-	; use_case_weights: optional, default is null. when true will scale influence weights by each case's weight_feature weight. if use_case_weights isn't specified, it will
-	;				   be true if auto ablation is enabled and false otherwise
-	; weight_feature: optional, default '.case_weight'.  name of feature whose values to use as case weights
-	; returns success when completed
 	#analyze
 	(declare
 		(assoc
 			;{type "list" values "string"}
+			;list of context features to analyze for
 			context_features (list)
 			;{type "list" values "string"}
+			;list of action features to analyze for. applicable only to 'single_targeted' mode
 			action_features (list)
 			;{type "number"}
+			;optional, (defaults to 1) number of cross validation folds to do. value of 1 does hold-one-out instead of k-fold
 			k_folds 1
 			;{type "boolean"}
+			;optional flag, default is false. if set to true will do k_folds ordered by session id indices, i.e., it'll hold out the
+			;	first 1/6th of the trained cases, then the next 1/6th, etc. in the order they were trained.
+			;	NOTE: if the dataset has been edited or has multiple trained sessions, the holdouts may be imbalanced due to multiple cases having
+			;	the same session id index or missing session indices, therefore this flag should only be used internally for validation
+			;{type "boolean"}
+			;optional
```

Comment:
can remove the extra type {} dict and optional here

Reply:
Corrected in 6d0f612